test/src/core/protocolGateway.spec.js

```diff
@@ -0,0 +1,228 @@
+/*****************************************************************************
+ * Copyright (c) 2021 Bosch.IO GmbH
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ ****************************************************************************/
+
+const NullLogger = require('../../helpers/null.logger');
+const importFresh = require('import-fresh');
+const mock = require('mock-require');
+const { prepareMockedMqttClient } = require('../../helpers/mock/mqtt.mock');
+const ProtocolGateway = require('../../../src/core/protocolGateway');
+
+describe('core.protocolGateway', () => {
+    let clientMock = null;
+
+    beforeAll(() => {
+        mock('../../../src/core/util/logger', NullLogger);
+    });
+
+    afterAll(() => {
+        mock.stop('../../../src/core/util/logger');
+    });
+
+    beforeEach(() => {
+        clientMock = prepareMockedMqttClient();
+    });
+
+    function createProtocolGateway(displayName, usePlatformProtocolOnly) {
+        // Clear cache, since Protocol Gateway reinstantiates the module
+        delete require.cache[require.resolve('../../../src/core/util/mqttClient')];
+        return new ProtocolGateway(importFresh('../../resources/protocolGateway.config.json'), displayName, usePlatformProtocolOnly);
+    }
+
+    it('It should publish messages to all defined adapters', async () => {
```

Comment:
Consistency: This file still uses these 'It should ...' pattern

Reply:
Sorry, I just have overseen that. Fixed it